src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceUnitTest.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.services;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.entities.TaskResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.cft.repository.TaskResourceRepository;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class CFTTaskDatabaseServiceUnitTest {
+
+    @Mock
+    private TaskResourceRepository taskRepository;
+
+    private CFTTaskDatabaseService cftTaskDatabaseService;
+
+    @BeforeEach
+    public void setUp() {
+        cftTaskDatabaseService = new CFTTaskDatabaseService(taskRepository);
+    }
+
+    @Test
+    void should_return_task_resource_by_specification() {
```

Comment:
can probably add a scenario where you get none back?

Reply:
The above scenario is covered in RWA-810. I have written based on AC.

Reply:
The AC is more at an e2e level e.g. for functional tests returning 2xx, 4xx, 5xx errors.  Covering those in another ticket is fine.
Unit tests however are covering individual units of code in isolation and should include checks on unexpected behaviour as good engineering practice even if the overall AC is not.  So I still think this should be covered in this PR.

Reply:
Happy to get chat and get @BenNzewi view on that.

Reply:
Added unit test

Reply:
👍